packages/utils/src/cloneObjectCyclic.ts

```diff
@@ -0,0 +1,37 @@
+// Makes a deep copy of an object, handling cyclical references.
```

Comment:
cc @marekrjpolak for review

Reply:
Yes, I did not want to replace it in case it cause some unexpected issues, since this PR is already doing quite a big changes. Maybe we can do it in future release.

Reply:
yes, please lets do this in a followup 👍